src/utilities/getAppBaseUrl.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { getAppGuiPath } from './getAppGuiPath'
+
+describe('getAppGuiPath', () => {
+  test.each([
+    ['https://example.com/gui', '/gui'],
+    ['https://example.com/gui/', '/gui'],
+    ['https://gui.example.com/gui/', '/gui'],
+    ['https://example.com/dev/gui', '/dev/gui'],
+    ['https://example.com/dev/gui/', '/dev/gui'],
+    ['https://gui.example.com/dev/gui/', '/dev/gui'],
+    ['https://example.com/guillermo/dev/gui', '/guillermo/dev/gui'],
+    ['https://example.com/guillermo/dev/gui/', '/guillermo/dev/gui'],
+    ['https://gui.example.com/guillermo/dev/gui/', '/guillermo/dev/gui'],
```

Comment:
what's the correct behaviour for `/gui/dev/gui/` ? should we match the last one or the first one? Can we add a test case for that?

Reply:
I’d say that’s a pretty arbitrary choice. I’ve added tests for matching the first segment.

Reply:
Isn't the last one safer option? I know it's pretty weird to have `/gui/something/gui/` but if we use the first one then the app would just show a 404 on the index load and all the links would "escape" that prefix right? If we use the last one then I guess we are good in that scenario right? Also how did this problem originate? (this might be a stupid question) Isn't just using relative links everywhere the solution to all of this?

Reply:
> I know it's pretty weird to have `/gui/something/gui/`

I don’t see why anything would ever link to `/gui/something/gui/`. That’s not something I think we need to protect against. If that should ever be a use case, I’d rather introduce a setting on the Kuma side that we discover via what’s proposed in https://github.com/kumahq/kuma/issues/3600#issuecomment-1308703684.

Reply:
> Also how did this problem originate? (this might be a stupid question) Isn't just using relative links everywhere the solution to all of this?

Relative URLs are a maintenance nightmare. If you’re at http://localhost:1234/a/b/c and you want to link to http://localhost:1234/d/e/f, the relative link target would need to be `../../../d/e/f`. It’s much more straightforward to have absolute or root-relative URLs that are the same no matter what page you’re on. For either, you need to know from where an application is served from. At the very least that means having a `<base href="http://localhost:1234/example/gui" />` (this makes all relative URL reference _relative_ to this base value) or an otherwise discoverable GUI base path.

Reply:
I think this is sufficiently tested now so I’m resolving this.